packages/slate/src/utils/path-utils.js

```diff
@@ -1,18 +1,17 @@
 import { List } from 'immutable'
 
 /**
- * Compare paths `path` and `b` to see which is before or after.
+ * Compare paths `path` and `target` to see which is before or after.
  *
  * @param {List} path
- * @param {List} b
+ * @param {List} target
  * @return {Number|Null}
  */
 
 function compare(path, target) {
-  // PERF: if the paths are not the same size we can exit early.
-  if (path.size !== target.size) return null
+  const m = min(path, target)
```

Comment:
A small suggestion.  I am feeling that perhaps it is a bit better to avoid `for`

```
let result = null;
path.zip(target).find( ([pv, tv] => {
   if (pv === tv) return false;
   result = pv > tv ? 1 : 0;
   return true
})

if (result !== null) return result;
return path.size === target.size ? 0 : null
```